### PR TITLE
take into account #spill/#unspill in sct checker

### PR DESCRIPTION
**Pull request by bgregoir**

### CODE REVIEW

File: compiler/src/sct_checker_forward.ml [hunk 1]
```diff
@@ -1412,6 +1458,12 @@ let init_constraint fenv f =
 
   let venv = List.fold_left do_local venv (Sv.elements (locals f)) in
 
+  let do_spill venv x =
+    let sx = Env.add_spill env x in
+    do_local venv sx in
+
+  let venv = List.fold_left do_spill venv (Sv.elements (spilled f)) in
```
Comment (vbgl) [inline compiler/src/sct_checker_forward.ml:1465]: Isn’t there a `Sv.fold` operator?

**Comment by vbgl** [comment]
Why don’t we just get rid of these fishy spill/unspill stuff?

File: compiler/src/sct_checker_forward.ml (hunk 1 above)
Comment (bgregoir) [reply compiler/src/sct_checker_forward.ml:1470]: yes I will change that.

Comment (bgregoir) [reply compiler/src/sct_checker_forward.ml]: done

**Comment by vbgl** [comment]
I really feel that this spill/unspill feature is poorly designed and that there is wide room for improvement.

**Pull request merged**